File: java-to-kotlin/src/main/java/nl/rabobank/kotlinmovement/recipes/model/IngredientRequest.java
package nl.rabobank.kotlinmovement.recipes.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public final class IngredientRequest {
    @NotBlank(message = "ingredient.name")
    private final String name;
    @NotNull(message = "ingredient.type")
    private final IngredientType type;
    @NotNull(message = "ingredient.weight")
    private final Integer weight;
}

File: java-to-kotlin-complete/src/main/kotlin/nl/alimeshkat/recipes/model/IngredientRequest.kt
package nl.alimeshkat.recipes.model

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class IngredientRequest(
    @field:NotBlank(message = "ingredient.name")
    val name: String?,
    @field:NotNull(message = "ingredient.type")
    val type: IngredientType?,
    @field:NotNull(message = "ingredient.weight")
    val weight: Int?
)
